### PR TITLE
Fix #34658 - Harmonize preseed templates

### PR DESCRIPTION
**Pull request by sbernhard**
Use preseed kernel options and use it in preseed templates
Added iPXE snapshots
Updated snapshots

### CODE REVIEW

File: app/views/unattended/provisioning_templates/iPXE/preseed_default_ipxe.erb [hunk 1]
```diff
@@ -38,7 +36,10 @@ ping --count 1 ${netX/dns} || echo Ping to DNS failed or ping command not availa
 <% kernel = boot_files_uris[0] -%>
 <% initrd = boot_files_uris[1] -%>
 
-kernel <%= kernel %> initrd=initrd.img interface=auto url=<%= foreman_url('provision')%><%= provision_url_suffix %> ramdisk_size=10800 root=/dev/rd/0 rw auto BOOTIF=01-${netX/mac:hexhyp} hostname=<%= @host.name %> domain=<%= @host.domain %> <%= keyboard_params %> <%= pxe_kernel_options %> locale=<%= host_param('lang') || 'en_US' %> <%= netcfg_args %>
+kernel <%= kernel %> initrd=initrd.img interface=auto url=<%= url %> ramdisk_size=10800 root=/dev/rd/0 rw auto BOOTIF=01-${netX/mac:hexhyp} <%= snippet("preseed_kernel_options").strip %>
+
 initrd <%= initrd %>
 
+imgstat
+sleep 2
```
Comment (ares) [inline app/views/unattended/provisioning_templates/iPXE/preseed_default_ipxe.erb:44]: are these two line related? could you perhaps add a comment on why we need this?

File: app/views/unattended/provisioning_templates/snippet/preseed_kernel_options.erb [hunk 2]
```diff
@@ -0,0 +1,55 @@
+<%#
+kind: snippet
+name: preseed_kernel_options
+model: ProvisioningTemplate
+snippet: true
+description: options for the kernel / preseed startup initialization
+-%>
```
Comment (ares) [inline app/views/unattended/provisioning_templates/snippet/preseed_kernel_options.erb:7]: It may be good to point out what host parameters this snippet considers since that information was dropped from the toplevel template. Or it may be good to add it to all those which use this snippet.

**Review by ares — CHANGES_REQUESTED** [review]
Just two small things, it looks like a good cleanup. I can give it a quick regression test but I didn't find anything goofy.

**Comment by ares** [comment]
FTR works fine with buster

File: app/views/unattended/provisioning_templates/iPXE/preseed_default_ipxe.erb (hunk 1 above)
Comment (sbernhard) [reply app/views/unattended/provisioning_templates/iPXE/preseed_default_ipxe.erb:44]: Well, I copied it from iPXE kickstart to be honest.  Is this ok?

File: app/views/unattended/provisioning_templates/snippet/preseed_kernel_options.erb (hunk 2 above)
Comment (sbernhard) [reply app/views/unattended/provisioning_templates/snippet/preseed_kernel_options.erb:7]: This doesn't even exist for kickstart and many different options exists which make it really hard to maintain. Therefore I removed it to make it similar to kickstart.

**Review by ares — APPROVED** [review]
Ok, makes sense. The two katello failures are unrelated (docker registry type). Merging, thanks @sbernhard!

**Comment by timdeluxe** [comment]
@sbernhard Please check my comment in https://projects.theforeman.org/issues/34658 - i think this PR breaked things.
